presto-main/src/main/java/com/facebook/presto/operator/TaskContext.java

```diff
@@ -325,6 +330,16 @@ public TaskStats getTaskStats()
             elapsedTime = new Duration(0, NANOSECONDS);
         }
 
+        synchronized (cumulativeMemoryStatsLock) {
```

Comment:
Oh wait, that's not going to work actually. For JOINs the memory is "owned" by the task (see `SharedLookupSource`), so I think we have to do it the way you have it